### PR TITLE
 2_4Grouping added for SNAP

### LINKED ISSUE #18180: 2_4 Grouping for CreateGroupingWorkspace
SNAP requests 2_4Grouping which is a modification of the East,West Groups option.

### PR DESCRIPTION
**Pull request by VickieLynch**
Description of work.

**To test:**

Compare new option with saved xml grouping file:

[SNAP_group_2_4.xml.txt](https://github.com/mantidproject/mantid/files/632050/SNAP_group_2_4.xml.txt)

```python
CreateGroupingWorkspace(InstrumentName='SNAP', GroupDetectorsBy='2_4Grouping', OutputWorkspace='group')
LoadDetectorsGroupingFile(InputFile='/home/vel/workspace/SNAP_group_2_4.xml', OutputWorkspace='loadedGroup')
```

Fixes #18180 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Algorithms/src/CreateGroupingWorkspace.cpp [hunk 1]
```diff
@@ -70,11 +70,13 @@ void CreateGroupingWorkspace::init() {
...
                   "If empty, then an empty GroupingWorkspace will be created.");
 
-  std::vector<std::string> grouping{"", "All", "Group", "Column", "bank"};
+  std::vector<std::string> grouping{"", "All", "Group", "2_4Grouping", "Column",
+                                    "bank"};
   declareProperty(
       "GroupDetectorsBy", "", boost::make_shared<StringListValidator>(grouping),
       "Only used if GroupNames is empty: All detectors as one group, Groups "
-      "(East,West for SNAP), Columns for SNAP, detector banks");
+      "(East,West for SNAP), 2_4Grouping for SNAP, Columns for SNAP, detector "
```
Comment (AndreiSavici) [inline Framework/Algorithms/src/CreateGroupingWorkspace.cpp]: The meaning of `2_4Grouping` is mentioned in the release notes, but not in the algorithm help

**Comment by AndreiSavici** [comment]
A comment that does not need to be addressed now: all the special cases are very SNAP centric

**Comment by VickieLynch** [comment]
@AndreiSavici I just tried all the options with POWGEN and only Column and 2_4Grouping do not work.  They result in empty workspaces.Maybe I should remove the East-West since Group works for the POWGEN grouping.

**Comment by VickieLynch** [comment]
Columns should work for POWGEN.  I will fix before this is merged.

**Comment by peterfpeterson** [comment]
It could be enforced in a `validateInputs`. Check if the instrument is SNAP, otherwise declare those two special groupings invalid.

**Comment by peterfpeterson** [comment]
This works as expected now. :shipit:

**Pull request merged**